### ISSUE TITLE
Add concept of "global variables"

### ISSUE THREAD

**Issue by JeanMertz** [blendle/automaat#24]
While working with some tasks, I noticed a pattern that there is a need to embed _secrets_ into tasks that shouldn't be provided by a task variable, but you also don't want to have exposed in the step definitions in your own Automaat repository.

For example, say I have a **SKU Sales Numbers** task that does a query to one of your internal APIs, based on a `{SKU}` variable. This API requires authentication by providing an `Authorization` header. You don't want the people using your task having to input this token (both for convenience _and_ security), so you want to bake this into the step definitions of your task.

However, you also want to be able to store your step definitions in source control, so that they are versioned, and can be tweaked by others, so hard-coding the token in your `tasks/sku_sales_numbers.json` file is out of the question.

You could resort to external templating systems that pre-process your task definition files to embed secrets when needed, but that adds extra overhead.

Instead, I think it makes sense to add a new concept to the _server_ component: **global variables**.

Basically, the following added functionality is needed:

* store global variables as key/value pairs in the database (both are strings)
* creating/updating global variables using GraphQL (no need to query, for now)
* support referencing global variables in step definitions (see #23)

I think it also makes sense to store these variables in encrypted form in the database. Not all of them are necessarily secrets, but I see no reason to distinguish between the two types of globals, and do see a need to store some of these values encrypted, and decrypt them on the fly, based on an encryption key provided on server start-up.
